tests/AppBundle/Controller/Admin/BlogControllerTest.php

```diff
@@ -27,35 +27,55 @@
  * Execute the application tests using this command (requires PHPUnit to be installed):
  *
  *     $ cd your-symfony-project/
- *     $ phpunit -c app
+ *     $ ./vendor/bin/phpunit
  */
 class BlogControllerTest extends WebTestCase
 {
-    public function testRegularUsersCannotAccessToTheBackend()
+    /**
+     * @dataProvider getUrlsForRegularUsers
+     */
+    public function testRegularUsers($httpMethod, $url, $statusCode)
     {
         $client = static::createClient([], [
             'PHP_AUTH_USER' => 'john_user',
             'PHP_AUTH_PW' => 'kitten',
         ]);
 
-        $client->request('GET', '/en/admin/post/');
+        $client->request($httpMethod, $url);
+        $this->assertEquals($statusCode, $client->getResponse()->getStatusCode());
+    }
 
-        $this->assertEquals(Response::HTTP_FORBIDDEN, $client->getResponse()->getStatusCode());
+    public function getUrlsForRegularUsers()
+    {
+        yield ['GET', '/en/admin/post/', Response::HTTP_FORBIDDEN];
+        yield ['GET', '/en/admin/post/1', Response::HTTP_FORBIDDEN];
+        yield ['GET', '/en/admin/post/1/edit', Response::HTTP_FORBIDDEN];
+        yield ['POST', '/en/admin/post/1/delete', Response::HTTP_FORBIDDEN];
     }
 
-    public function testAdministratorUsersCanAccessToTheBackend()
+    /**
+     * @dataProvider getUrlsForAdminUsers
+     */
+    public function testAdminUsers($httpMethod, $url, $statusCode)
     {
         $client = static::createClient([], [
             'PHP_AUTH_USER' => 'anna_admin',
             'PHP_AUTH_PW' => 'kitten',
         ]);
 
-        $client->request('GET', '/en/admin/post/');
+        $client->request($httpMethod, $url);
+        $this->assertEquals($statusCode, $client->getResponse()->getStatusCode());
+    }
 
-        $this->assertEquals(Response::HTTP_OK, $client->getResponse()->getStatusCode());
+    public function getUrlsForAdminUsers()
+    {
+        yield ['GET', '/en/admin/post/', Response::HTTP_OK];
+        yield ['GET', '/en/admin/post/1', Response::HTTP_OK];
+        yield ['GET', '/en/admin/post/1/edit', Response::HTTP_OK];
+        yield ['POST', '/en/admin/post/1/delete', Response::HTTP_FOUND];
```

Comment:
This test no change the DB status for now, because this redirect come from [here](https://github.com/javiereguiluz/symfony-demo/blob/4cfce12ffd912e31fcffde65737c37de1c3f8fb3/src/AppBundle/Controller/Admin/BlogController.php#L181)